graphql/resolvers.go

```diff
@@ -3231,8 +3233,18 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 		}
 		mainlineCommitVersion := MainlineCommitVersion{}
 
+		shouldNotBeRolledUp := true
+		if buildVariantOptions != nil {
+			if utility.FromBoolPtr(v.Activated) && (len(buildVariantOptions.Statuses) != 0 || len(buildVariantOptions.Tasks) != 0 || len(buildVariantOptions.Variants) != 0) {
+				activatedCommitsCheckedCount++
```

Comment:
Is this just for debugging purposes? I think it might be easier to move all of the length checks into `hasMatchingTasks` to simplify things

Reply:
Yeah it is, This collects the amount of activated mainline commits we have checked. I think its better to keep them separate since the values aren't really related. The only way they are related is that `hasMatchingTasks` coincidentally gets run when ever we increment this value.